inc/namespace.php

```diff
@@ -125,88 +132,142 @@ function remove_yoast_submenu_page() {
+	}
+
+	return $options;
 }
 
 /**
- * Update twitter metadata to use Tachyon img URL.
+ * Check if we should override the metadata options.
  *
- * @param array $meta Twitter metadata.
+ * Compares the SEO metadata options saved in the config file with the default values. If anything has been changed, returns true.
  *
- * @return array Twitter metadata with image using Tachyon URL, if any.
+ * @return bool True if metadata values have been saved.
  */
-function metadata_img_as_tachyon_twitter( array $meta ) : array {
-	return metadata_img_as_tachyon( $meta, [
-		'resize' => '1200,600', // crop.
-	] );
+function should_override_metadata_options() : bool {
+	$config = Altis\get_config()['modules']['seo']['metadata'] ?? [];
+	$default_config = apply_filters( 'altis.config.default', [] )['modules']['seo']['metadata'];
+
+	// If the config matches the default, we aren't overriding metadata options.
+	if ( empty( config_diff( $config, $default_config ) ) ) {
+		return false;
+	}
+
+	// If any changes have been made to the metadata config, they will override the default options.
+	return true;
 }
 
 /**
- * Update opengraph metadata to use Tachyon img URL.
+ * Determine the difference between two associative arrays by serializing the arrays and comparing them.
+ *
+ * @link https://stackoverflow.com/a/22355153/11710741
  *
- * @param array $meta opengraph metadata.
+ * @param array $config The array to compare, e.g. a config value.
+ * @param array $default_config The array to compare against, e.g. the default config.
  *
- * @return array opengraph metadata with image using Tachyon URL, if any.
+ * @return array The difference between the two arrays.
  */
-function metadata_img_as_tachyon_opengraph( array $meta ) : array {
-	return metadata_img_as_tachyon( $meta, [
-		'fit' => '1200,627', // no crop.
-	] );
+function config_diff( array $config, array $default_config ) : array {
+	return array_map( 'unserialize',
+		array_diff(
+			array_map( 'serialize', $config ),
+			array_map( 'serialize', $default_config )
+		)
+	);
 }
 
 /**
- * Update metadata image URL to use Tachyon URL with specified image settings.
+ * Get a WordPress image ID by the URL.
  *
- * @param array $meta          Metadata per social media type.
- * @param array $img_settings Image settings: size and crop to be used in Tachyon URL.
+ * @param string $url URL of the uploaded image.
  *
- * @return array Metadata with updated image URL using Tachyon, if an image is specified.
+ * @return int|string Either the ID of the image or an empty string if no matching ID was found.
  */
-function metadata_img_as_tachyon( array $meta, array $img_settings = [] ) : array {
-	// Stop - no image for metadata.
-	if ( ! isset( $meta['image'] ) ) {
-		return $meta;
-	}
+function get_image_id_from_url( string $url ) {
+	global $wpdb;
+	$url = preg_replace( '/-\d+x\d+(?=\.(jpg|jpeg|png|gif)$)/i', '', $url );
+
+	$image = $wpdb->get_col( $wpdb->prepare(
+		"SELECT ID FROM $wpdb->posts WHERE guid=%s;",
```

Comment:
This won't be multisite aware and could end up running twice on every page load, the guid isn't a reliable indicator of the attachment URL and additionally this query isn't cached.

There's a built in WP function `attachment_url_to_postid()` too but that's also not multisite aware. Can we avoid doing this at all? Will Yoast freak out if it doesn't know the ID?

Reply:
I will test without. I don't _think_ there's any risk but I know that when you save the settings normally on the settings page, it pulls from the media library specifically and therefore stores the ID.